src/Simulation/CsharpGeneration/SimulationCode.fs

```diff
@@ -69,16 +78,24 @@ module SimulationCode =
     let prependNamespaceString (name : QsQualifiedName) =
         name.Namespace.Replace (".", "__") + "__" + name.Name
 
+    let isConcreteIntrinsic (context:CodegenContext) =
+        match context.assemblyConstants.TryGetValue AssemblyConstants.GenerateConcreteIntrinsic with
+        | true, "false" -> false
+        | true, _ -> true
+        | false, _ -> false
+
     let needsFullPath context (op:QsQualifiedName) =
         let hasMultipleDefinitions() = if context.byName.ContainsKey op.Name then context.byName.[op.Name].Length > 1 else false
         let sameNamespace = match context.current with | None -> false | Some n -> n.Namespace = op.Namespace
 
+        let namespaces = if isConcreteIntrinsic context then autoNamespacesWithInterfaces else autoNamespaces
```

Comment:
As written, this might have that effect, but I don't think it should be required. Generating the concrete intrinsics with calls into the interfaces and generating the interfaces themselves can be done at the same time while following the same set of assumptions. As long as all the files exist before the C# is compiled then it will be fine; the concrete intrinsics and the interfaces don't actually need any information other than the Q# declarations and don't depend on any generated details from each other.